models/ssh_key.go

```diff
@@ -1030,3 +1058,225 @@ func SearchDeployKeys(repoID int64, keyID int64, fingerprint string) ([]*DeployK
 	}
 	return keys, x.Where(cond).Find(&keys)
 }
+
+// PrincipalKey represents principal key information.
+type PrincipalKey struct{}
+
+// AddPrincipalKey adds new principal to database and authorized_principals file.
+func AddPrincipalKey(ownerID int64, name, content string, loginSourceID int64) (*PublicKey, error) {
+	log.Trace(content)
+
```

Comment:
I suspect that this is a debugging command that we won't want in the final version.

either delete it:

```suggestion
```

Or:

```suggestion
	log.Trace("AddPrincipalKey for %d on login %d with name %s: %s", ownerID, loginSourceID, name, content)

```